In [1]:
pip install Datascrubber


   ---------------------------------------- 0.0/11.0 MB ? eta -:--:--
   ---------------------------------------- 0.0/11.0 MB ? eta -:--:--
   ---------------------------------------- 0.0/11.0 MB ? eta -:--:--
   ---------------------------------------- 0.0/11.0 MB ? eta -:--:--
   ---------------------------------------- 0.0/11.0 MB ? eta -:--:--
   ---------------------------------------- 0.0/11.0 MB ? eta -:--:--
   ---------------------------------------- 0.0/11.0 MB ? eta -:--:--
   ---------------------------------------- 0.0/11.0 MB ? eta -:--:--
   ---------------------------------------- 0.0/11.0 MB ? eta -:--:--
   ---------------------------------------- 0.0/11.0 MB ? eta -:--:--
   ---------------------------------------- 0.0/11.0 MB ? eta -:--:--
   ---------------------------------------- 0.0/11.0 MB ? eta -:--:--
   ---------------------------------------- 0.0/11.0 MB ? eta -:--:--
   ---------------------------------------- 0.0/11.0 MB ? eta -:--:--
   ----------------

ERROR: Exception:
Traceback (most recent call last):
  File "c:\Users\Tonny\AppData\Local\Programs\Python\Python312\Lib\site-packages\pip\_vendor\urllib3\response.py", line 438, in _error_catcher
    yield
  File "c:\Users\Tonny\AppData\Local\Programs\Python\Python312\Lib\site-packages\pip\_vendor\urllib3\response.py", line 561, in read
    data = self._fp_read(amt) if not fp_closed else b""
           ^^^^^^^^^^^^^^^^^^
  File "c:\Users\Tonny\AppData\Local\Programs\Python\Python312\Lib\site-packages\pip\_vendor\urllib3\response.py", line 527, in _fp_read
    return self._fp.read(amt) if amt is not None else self._fp.read()
           ^^^^^^^^^^^^^^^^^^
  File "c:\Users\Tonny\AppData\Local\Programs\Python\Python312\Lib\site-packages\pip\_vendor\cachecontrol\filewrapper.py", line 98, in read
    data: bytes = self.__fp.read(amt)
                  ^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Tonny\AppData\Local\Programs\Python\Python312\Lib\http\client.py", line 479, in read
    s = self.fp.read

   ------------------ --------------------- 5.0/11.0 MB 20.3 kB/s eta 0:04:56
   ------------------ --------------------- 5.0/11.0 MB 20.3 kB/s eta 0:04:56
   ------------------ --------------------- 5.0/11.0 MB 20.3 kB/s eta 0:04:56
   ------------------ --------------------- 5.0/11.0 MB 20.3 kB/s eta 0:04:56
   ------------------ --------------------- 5.0/11.0 MB 20.3 kB/s eta 0:04:56
   ------------------ --------------------- 5.0/11.0 MB 20.3 kB/s eta 0:04:56
   ------------------ --------------------- 5.0/11.0 MB 20.3 kB/s eta 0:04:56
   ------------------ --------------------- 5.0/11.0 MB 20.3 kB/s eta 0:04:56
   ------------------ --------------------- 5.0/11.0 MB 20.3 kB/s eta 0:04:56
   ------------------ --------------------- 5.0/11.0 MB 20.3 kB/s eta 0:04:56
   ------------------ --------------------- 5.0/11.0 MB 20.3 kB/s eta 0:04:56
   ------------------ --------------------- 5.0/11.0 MB 20.3 kB/s eta 0:04:56
   ------------------ --------------------- 5.0/11.0 MB 20.3 kB/

In [17]:
import pandas as pd

In [2]:
loan = pd.read_csv("Loan_Approval_Data.csv")
loan.head()

NameError: name 'pd' is not defined

### 1. The marginal probability of any continuous variable

In [19]:
#Creating bins
bins = [0,2000,4000,6000,8000,10000,20000,30000,40000,50000]
labels = ['0-2000','2000-4000','4000-6000','6000-8000','8000-10000','10000-20000','20000-30000','30000-40000','40000-50000']

#Creating a new column for ApplicantIncome_Bin
loan['ApplicantIncome_bin'] = pd.cut(loan['ApplicantIncome'], bins = bins, labels=labels, right=False)

##Calculating the Marginal Probabilities
ApplicantIncome_probabilities = loan['ApplicantIncome_bin'].value_counts(normalize=True).sort_index()
ApplicantIncome_probabilities

ApplicantIncome_bin
0-2000         0.065466
2000-4000      0.476268
4000-6000      0.229133
6000-8000      0.096563
8000-10000     0.047463
10000-20000    0.070376
20000-30000    0.008183
30000-40000    0.006547
40000-50000    0.000000
Name: proportion, dtype: float64

### 2.Marginal Probability for any categorical variable


In [20]:
Self_employed_marginal = loan['Self_Employed'].value_counts()/len(loan)
print(f"Marginal probabilities for Self_Employed are : \n {Self_employed_marginal}")


Marginal probabilities for Self_Employed are : 
 Self_Employed
No     0.814332
Yes    0.133550
Name: count, dtype: float64


### 3.Probability that a "female" applicant has her Loan_Status reflecting "Y"

In [21]:
#Filtering the data for only female applicants whose loan status is Y
filtered_data = (loan['Gender'] == 'Female') & (loan['Loan_Status'] == 'Y')

#Calculating number of outcomes from the filtered data
Female_Y_count = loan[filtered_data]['Gender'].count()
Female_Y_count

#Calculating the probability for females with a loan status reflecting Y
Female_Y_probability=Female_Y_count/len(loan)
Female_Y_probability

#Calculating the Y probability
Y_count=loan[loan['Loan_Status']=="Y"]['Loan_Status'].count()
Y_probability=Y_count/len(loan)

#Calculating the final conditional probability of females given a loan status reflecting Y
Final_probability=Female_Y_probability/Y_probability
Final_probability
print(f"The probability of females with a loan status reflecting Y is {Final_probability}")

The probability of females with a loan status reflecting Y is 0.17772511848341233


### 4.Explain the central tendency for any two continuous variables.

In [22]:
loan.describe()

ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
count       614.000000         614.000000  592.000000         600.00000   
mean       5403.459283        1621.245798  146.412162         342.00000   
std        6109.041673        2926.248369   85.587325          65.12041   
min         150.000000           0.000000    9.000000          12.00000   
25%        2877.500000           0.000000  100.000000         360.00000   
50%        3812.500000        1188.500000  128.000000         360.00000   
75%        5795.000000        2297.250000  168.000000         360.00000   
max       81000.000000       41667.000000  700.000000         480.00000   

       Credit_History  
count      564.000000  
mean         0.842199  
std          0.364878  
min          0.000000  
25%          1.000000  
50%          1.000000  
75%          1.000000  
max          1.000000

Applicant Income<br>
Mean=5403.459283<br>
Median=3812.500000<br>

Loan Amount<br>
Mean=146.412162<br>
Median=128.000000<br>

### 5.Explain the relationship between the variables "Gender" and "Loan Status"

In [23]:
#Running the chisquare test to first find out the relationship

#Generating a cross-tabulation between 'Gender' and 'Loan Status'
Cross_tab=pd.crosstab(index=loan['Gender'],columns=loan['Loan_Status'])
print(Cross_tab)

from scipy.stats import chi2_contingency
ChiSqResult = chi2_contingency(Cross_tab)
print('The P-Value is:', ChiSqResult[1])

Loan_Status    N    Y
Gender               
Female        37   75
Male         150  339


ModuleNotFoundError: No module named 'scipy'

Since the P-Value >0.05 then Gender and Loan_Status are not correlated,